app/models/stop.rb

```diff
@@ -116,12 +115,39 @@ def before_destroy_making_history(changeset, old_model)
   has_many :routes_serving_stop
   has_many :routes, through: :routes_serving_stop
 
+  def operators_serving_stop_and_platforms
+    OperatorServingStop
+      .where('stop_id IN (?) OR stop_id = ?', Stop.where(parent_stop_id: self.id).select(:id), self.id)
+      .select(:operator_id)
+      .uniq
+      .map { |o| OperatorServingStop.new(stop: self, operator_id: o.operator_id) }
```

Comment:
@irees Would the following work to return the same results, using only the DB:
```ruby
    OperatorServingStop
      .where('stop_id IN (?) OR stop_id = ?', Stop.where(parent_stop_id: self.id).select(:id), self.id)
      .select('DISTINCT ON (current_operators_serving_stop.operator_id) *')
```
?